scipy/stats/_qmc.py

```diff
@@ -243,80 +254,54 @@ def discrepancy(sample, iterative=False, method='CD'):
     0.008142039609053513
 
     """
-    sample = np.asarray(sample)
+    sample = np.asarray(sample, dtype=np.float64, order="C")
 
     # Checking that sample is within the hypercube and 2D
     if not sample.ndim == 2:
-        raise ValueError('Sample is not a 2D array')
+        raise ValueError("Sample is not a 2D array")
 
     if not (np.all(sample >= 0) and np.all(sample <= 1)):
-        raise ValueError('Sample is not in unit hypercube')
+        raise ValueError("Sample is not in unit hypercube")
+
+    if not isinstance(workers, int):
+        raise ValueError(
+            f"workers param needs to be an int. Found " f"{type(workers)!r}"
+        )
+
+    if workers == -1:
+        workers = os.cpu_count()
+        if workers is None:
+            raise NotImplementedError(
+                'Cannot determine the number of cpus using os.cpu_count(), '
+                'cannot use -1 for the number of workers')
+    elif workers <= 0:
+        raise ValueError(f'Invalid number of workers: {workers}, must be -1 '
+                         'or > 0')
+
+    if method == "CD":
+        return _cy_wrapper_centered_discrepancy(
```

Comment:
maybe a more idiomatic way of writing this is:
```
methods = {
    'CD' : _cy_wrapper_centered_discrepancy,
    'WD' : _cy_wrapper_wrap_around_discrepancy,
    ...
}
# check method is one of the expected values here, raise ValueError if not

return methods[method](sample, iterative, workers=workers)
```

Reply:
done in 6e47895e17279ad78ed582d1bd18de7368604df2